pkg/tbtc/storage.go

```diff
@@ -0,0 +1,164 @@
+package tbtc
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"encoding/hex"
+	"fmt"
+	"github.com/keep-network/keep-common/pkg/persistence"
+	"sync"
+)
+
+// walletStorage is the component that persists data of the wallets managed
+// by the given node. All functions of the storage are safe for concurrent use.
+type walletStorage struct {
```

Comment:
I think we should separate the in-memory cache from the storage layer. We had this separation in [v1 ECDSA implementation](https://github.com/keep-network/keep-ecdsa/tree/main/pkg/registry) and we have this separation in [v1/v2 Beacon implementation](https://github.com/keep-network/keep-core/tree/main/pkg/beacon/registry). 

Right now the implementation is simple, but we must add snapshots and logic to clean up the storage from wallets that should be archived. Once we do that, without the separation, we will end up with a structure responsible for calling disk read/writes, marshalling/unmarshalling, fetching information from the chain, and maintaining an in-memory cache. This is way too much responsibility in a single structure.

I am fine with having two files or just one Go file, but I really think we should have two structures:
- one handling in-memory cache and in the future calling the chain to clean up the local storage,
- one handling marshaling/unarshaling and interacting with disk IO API.

Reply:
See 39a977d3